portal/app/[locale]/staking-dashboard/_components/stakeTable/lockupTime.tsx

```diff
@@ -1,15 +1,41 @@
 import { DurationTime } from 'components/durationTime'
+import Skeleton from 'react-loading-skeleton'
+
+import { useCalculateApy } from '../../_hooks/useCalculateApy'
 
 type Props = {
   lockupTime: bigint
+  tokenId: string
 }
 
-export const LockupTime = function ({ lockupTime }: Props) {
+export const LockupTime = function ({ lockupTime, tokenId }: Props) {
   const seconds = Number(lockupTime)
+  const {
+    data: apy,
+    error,
+    isLoading,
+  } = useCalculateApy({ tokenId: BigInt(tokenId) })
+
+  const renderApy = function () {
+    if (isLoading) {
+      return <Skeleton className="h-4 w-16" />
+    }
+
+    if (error || apy === undefined) {
+      return <span className="text-xs font-normal text-neutral-500">-</span>
+    }
+
+    return (
+      <span className="text-xs font-normal text-emerald-600">{apy}% APY</span>
```

Comment:
`APY` is `TEA` in spanish. So this probably should be translated. Check the resource files because I think there are other places where we do this translation already

Reply:
In addition to that, although unlikely for now, you may want to consider adding formatting for large numbers, in case we ever get an APY that has decimals or is larger than 999